### ISSUE TITLE
Kernel Deadlock

### ISSUE THREAD

**Issue by JeremyWildsmith** [JeremyWildsmith/DCPUOS#2]
A deadlock can occur in signal_poll or keyboard_poll called through the Kernel Service Provider thread.

Consider signal_poll:

````
;All signal handlers are threading locked. This is because they may result in, for example, their address space being deallocated.
;If they are locked, they have time to return before their address space is overwritten.
signal_poll:
	call threading_lock
	signal_poll_start:
	ife [signal_queue_currentOffset], 0xFFFF
	jmp signal_poll_end
	
	set a, [signal_queue_currentOffset]
	
	set x, [a + signal_queue_processes]
	set x, [x + PEIMAGE_OFFSET_RVA_SIGHANDLER]
	_push [a + signal_queue_codes]
	
	ifn x, 0
		call x
	
	add sp, 1
	
	sub [signal_queue_currentOffset], 1
	jmp signal_poll_start
	
	signal_poll_end:
	call threading_free
	ret
````

A deadlock appears to be occurring between signal_poll and keyboard_poll when tested on dcpu-ide. I am still not entirely sure why.

Strangely the kernel service thread is not executing at all after the clock interrupt causes the threading system to switch threads, despite not using mutex objects at all. 

The thread that seems to be executing when the kernel is locked up is stuck hanging in mutex_lock waiting for a mutex to be released.

**Comment by JeremyWildsmith** [comment]
Looked into this more, the deadlock occurs when a hardware interrupt is raised during kernel_initThreadEp inside kernel_initThreadEp

**Comment by JeremyWildsmith** [comment]
Increasing clock interrupt interval for once every 60th of a second allows me to reproduce this in dev-cpu.

**Comment by JeremyWildsmith** [comment]
Resolved in 6becc580e189ffe73bb0ab9dd29b156154a613db
